lib/engine/auto_router.rb

```diff
@@ -63,12 +67,24 @@ def compute(corporation, **opts)
             end
           end
 
+          assign_both = lambda do |a, b|
+            if a == last_left || b == last_right
+              left = b
+              right = a
+            else
+              left = a
+              right = b
+            end
+            complete.call
+          end
+
           paths.each do |path|
             chain << path
             a, b = path.nodes
```

Comment:
maybe this can be cleaned up with 

assign.call(a, b) if a || b

Reply:
No, that will still call if a and b are true.

Reply:
you'll have to rewrite the assigns methods

Reply:
Yep.